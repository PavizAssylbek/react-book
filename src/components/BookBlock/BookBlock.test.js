import React from 'react'
import BookBlock from './index'

const onClick = () => {
  console.log("check function")
}



describe('should render BookBlock component', () => {
  
  it('should contain function onClick={Func}', () => {
    const component = shallow(<BookBlock onClickAddPizza={onClick} />)
    expect(component).toMatchSnapshot()
  });

  it('should contain id', () => {
    const component = shallow(<BookBlock id={1} />)
    expect(component).toMatchSnapshot()
  });

  it('should contain name ', () => {
    const component = shallow(<BookBlock name="string" />)
    expect(component).toMatchSnapshot()
  });

  it('should contain img ', () => {
    const component = shallow(<BookBlock imageUrl="string" />)
    expect(component).toMatchSnapshot()
  });

  it('should contain price and addedCount ', () => {
    const component = shallow(<BookBlock price={123} addedCount={1234} />)
    expect(component).toMatchSnapshot()
  });

  it('should contain function onClickAddPizza ', () => {
    const component = shallow(<BookBlock onClickAddPizza={onClick} />)
    expect(component).toMatchSnapshot()
  });

})