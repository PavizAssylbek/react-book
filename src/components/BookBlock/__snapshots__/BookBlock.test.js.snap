// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render BookBlock component should contain function onClick={Func} 1`] = `ShallowWrapper {}`;

exports[`should render BookBlock component should contain function onClickAddPizza  1`] = `ShallowWrapper {}`;

exports[`should render BookBlock component should contain id 1`] = `ShallowWrapper {}`;

exports[`should render BookBlock component should contain img  1`] = `ShallowWrapper {}`;

exports[`should render BookBlock component should contain name  1`] = `ShallowWrapper {}`;

exports[`should render BookBlock component should contain price and addedCount  1`] = `ShallowWrapper {}`;
